### PR TITLE
merge bitcoin advanced transactions into master 

### PR DESCRIPTION
**Pull request by najla-quant**

### CODE REVIEW

**Comment by lukerQuant** [comment]
Checked this before on the merge into develop

**Review by lukerQuant — APPROVED** [review]

File: examples/bitcoin-scripts/p2sh/htlc/cancel-p2sh-payment-channel.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      console.log('Your ' + result.dltData[counter].dlt + ' value transfer transaction hash is: ' + result.dltData[counter].transactionHash);
+      console.log("");
+      counter = counter + 1;
+    }
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
+
+// non-final until the lock time is reached
```
Comment (alexchiriac) [inline examples/bitcoin-scripts/p2sh/htlc/cancel-p2sh-payment-channel.js:124]: Missing new line here although not a big problem hehe

File: examples/bitcoin-scripts/p2sh/htlc/create-p2sh-payment-channel.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    console.log(`currentContractCode ${currentContractCode.toString('hex')}`);
+    const p2shPaymentChannel = createHashTimeLockContractPaymentChannel(currentContractCode, TransactionBitcoinScriptTypeOptions.P2SH, overledger.dlts.bitcoin.addressType);
+    console.log(`p2sh ${JSON.stringify(p2shPaymentChannel)}`);
+    console.log(`p2sh address: ${p2shPaymentChannel.address}`);
+    console.log(`p2sh output script: ${p2shPaymentChannel.output.toString('hex')}`);
+    console.log(`p2sh redeem script: ${p2shPaymentChannel.redeem.output.toString('hex')}`);
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
```
Comment (alexchiriac) [inline examples/bitcoin-scripts/p2sh/htlc/create-p2sh-payment-channel.js:62]: Same here

File: examples/bitcoin-scripts/p2wsh/htlc/create-p2wsh-payment-channel.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    const currentContractCode = generateHashTimeLockContractCodeSHA256(claimPublicKey, refundPublicKey, hashSecret, timeLock);
+    const p2wshPaymentChannel = createHashTimeLockContractPaymentChannel(currentContractCode, TransactionBitcoinScriptTypeOptions.P2WSH, overledger.dlts.bitcoin.addressType);
+    console.log(`p2wsh ${JSON.stringify(p2wshPaymentChannel)}`);
+    console.log(`p2wsh address: ${p2wshPaymentChannel.address}`);
+    console.log(`p2wsh output script: ${p2wshPaymentChannel.output.toString('hex')}`);
+    console.log(`p2wsh witness script: ${p2wshPaymentChannel.redeem.output.toString('hex')}`);
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
```
Comment (alexchiriac) [inline examples/bitcoin-scripts/p2wsh/htlc/create-p2wsh-payment-channel.js:55]: Same here

File: examples/bitcoin-scripts/p2wsh/multisig/fund-p2wsh-multisig.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    counter = 0;
+    while (counter < result.dltData.length){
+      console.log('Your ' + result.dltData[counter].dlt + ' value transfer transaction hash is: ' + result.dltData[counter].transactionHash);
+      console.log("");
+      counter = counter + 1;
+    }
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
```
Comment (alexchiriac) [inline examples/bitcoin-scripts/p2wsh/multisig/fund-p2wsh-multisig.js:113]: And here

File: examples/bitcoin-segwit/a-to-b-transaction-fund-nested-segwit.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      console.log("");
+      counter = counter + 1;
+    }
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
+
+
+// https://blockstream.info/testnet/tx/3dfa35edfaa30857c44dad7bb948c582d3a563ed758c7fd8d6d918c562051bf7
```
Comment (alexchiriac) [inline examples/bitcoin-segwit/a-to-b-transaction-fund-nested-segwit.js:114]: Aaand here..

File: examples/bitcoin-segwit/a-to-b-transaction-native-segwit.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      console.log("");
+      counter = counter + 1;
+    }
+  } catch (e) {
+    console.error('error:', e);
+  }
+})();
+
+
+// https://blockstream.info/testnet/tx/3dfa35edfaa30857c44dad7bb948c582d3a563ed758c7fd8d6d918c562051bf7
```
Comment (alexchiriac) [inline examples/bitcoin-segwit/a-to-b-transaction-native-segwit.js:103]: Same

File: packages/overledger-types/src/Account.ts [hunk 7]
```diff
@@ -16,8 +20,12 @@ type Account = {
...
+  isSegwit?: boolean,
+  isNestedSegwit?: boolean,
+  script?: string,
+  redeemScript?: string,
   provider?: string,
+  password?: string
 };
 
-export default Account;
+export default Account;
```
Comment (alexchiriac) [inline packages/overledger-types/src/Account.ts:31]: And here

**Review by alexchiriac — APPROVED** [review]
LGTM! Linting is up to you :)

**Review by kashif-quant — COMMENTED** [review]
ok

**Pull request merged**